libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -114,6 +114,7 @@ internal class MapboxTripSession(
             }
             is SetRoutes.RefreshRoutes -> {
                 if (routes.isNotEmpty()) {
+                    isUpdatingRoute = true
```

Comment:
According to https://github.com/mapbox/mapbox-navigation-android/pull/6397#discussion_r982144756 we don't need to ignore status updates while routes are being refreshed. Also we don't do that in #6721.

Reply:
In #6721 we ignore status updates for all reasons except alternative. So we ignore for refresh.
We might get (and we get) route refresh updates for refreshed route **before** `RouteRefreshCallback` is invoked. Which is wrong because the user might receive `RouteProgress` update for the route which they have not yet received in `RoutesObserver`.
I know that in the comment it says:
> Refreshed route cannot be provided by a developer so they are not even aware that it's coming, so it's also safe to keep going.

but maybe there is more to that. As far as I understand, we guarantee that first the route will be delivered via `RoutesObserver`, then the status via `RouteProgressObserver`. So technically if I revert this PR the check `if (it.navigationRoute != currentRoute)` in `RouteProgressObserver` might evaluate to true, I'm not sure if it's OK. @LukasPaczos WDYT?

Reply:
> In #6721 we ignore status updates for all reasons except alternative. So we ignore for refresh. 

You probably mean #6715? Because in #6721 status updates are ignored only if `setRoutesInfo` is an instance of `BasicSetRoutesInfo`, which doesn't include refreshes.

Reply:
I do, I thought they were the same.

Reply:
I think https://github.com/mapbox/mapbox-navigation-android/pull/6397#discussion_r982144756 and https://github.com/mapbox/mapbox-navigation-android/pull/6397#discussion_r982292610 are incorrect in the end. If Nav Native would invoke a new status callback synchronously when refreshing routes, before native refresh callback returns, we'd provide to developers a new progress update that has the old route but all progress values are already based on the refreshed route. That's why we need to pause updates until native refresh callback returns.

Reply:
I don't think we considered a synchronous status generation when evaluating https://github.com/mapbox/mapbox-navigation-android/pull/6397, which we now know can happen and does happen for multiple APIs.

Reply:
Maybe we need to fix `release-v2.9` branch then to do the same. But I don't know how critical that issue is.

Reply:
That's a pretty marginal case I'd imagine, I don't think it needs a dedicated patch release.